protobuf/v1/nexus.proto

```diff
@@ -67,6 +70,18 @@ message CreateNexusRequest {
   NexusNvmePreemption preemptPolicy = 11; // the preemption policy to use
 }
 
+// Create nexus snapshot.
+message  CreateSnapshotRequest {
+  string nexus_uuid = 1;    // UUID of the nexus
+  string snapshot_name = 2; // User-specific snapshot name.
```

Comment:
Should the name be optional? If a user isn't coming up with a name, we internally create a name as a string concat of other params and timestamp for example.

Reply:
> Should the name be optional? If a user isn't coming up with a name, we internally create a name as a string concat of other params and timestamp for example.

I think name should be generated by control plane if the information is not coming as part of the CSI gRPC. For dataplane, we can consider the parameter with valid value.

Reply:
Alright, yes since this is gRPC spec between control-plane and data-plane, control-plane should take are of that.